sql/analyzer/scope.go

```diff
@@ -30,6 +30,9 @@ type Scope struct {
 	memos []sql.Node
 	// recursionDepth tracks how many times we've recursed with analysis, to avoid stack overflows from infinite recursion
 	recursionDepth int
+	// CurrentNodeIsSubqueryExpression is true when the last scope (i.e. the most inner of the outer scope levels) has been
+	// created by a subquery expression. This is needed in order to calculate outer scope visibility for derived tables.
+	CurrentNodeIsSubqueryExpression bool
```

Comment:
Worried about this extra bit of state -- who's responsible for setting it back to false?

Reply:
The `Scope.newScope` method handles creating a new subscope from the current scope and that method does not set this flag on the new subscope, so it gets correctly cleared for each new subscope. 